pilot/test/integration/infra.go

```diff
@@ -643,25 +715,46 @@ func (infra *infra) createSidecarInjector() error {
 
 	// webhook deployment
 	infra.CABundle = base64.StdEncoding.EncodeToString(ca)
-	if yaml, err := fill("sidecar-injector.yaml.tmpl", infra); err != nil { // nolint: vetshadow
+	if filledYaml, err := infra.fill("sidecar-injector.yaml.tmpl", infra); err != nil { // nolint: vetshadow
 		return err
-	} else if err = infra.kubeApply(yaml, infra.IstioNamespace); err != nil {
+	} else if err = infra.kubeApply(filledYaml, infra.IstioNamespace); err != nil {
 		return err
 	}
 
 	// wait until injection webhook service is running before
 	// proceeding with deploying test applications
-	if _, err = util.GetAppPods(client, kubeconfig, []string{infra.IstioNamespace}); err != nil {
+	if _, err = util.GetAppPods(infra.kubeClient, infra.KubeConfig, []string{infra.IstioNamespace}); err != nil {
 		return fmt.Errorf("sidecar injector failed to start: %v", err)
 	}
 	return nil
 }
 
 func (infra *infra) deleteSidecarInjector() {
-	if yaml, err := fill("sidecar-injector.yaml.tmpl", infra); err != nil {
+	if filledYaml, err := infra.fill("sidecar-injector.yaml.tmpl", infra); err != nil {
 		log.Infof("Sidecar injector template could not be processed, please delete stale injector webhook: %v",
 			err)
-	} else if err = infra.kubeDelete(yaml, infra.IstioNamespace); err != nil {
+	} else if err = infra.kubeDelete(filledYaml, infra.IstioNamespace); err != nil {
 		log.Infof("Sidecar injector could not be deleted: %v", err)
 	}
 }
+
+// fill a file based on a template
```

Comment:
As an aside, I wonder if test code would be more maintainable if we created the k8s resources directly with client-go instead of some combination of `text/template` and shelling out to kubectl. For example, see the configmap creation for mutating webhooks below. It's immediately clear what test values are used during resource creation vs. applying a template over the `infra` object which selectively uses some (but not all) of the fields.

https://github.com/istio/istio/blob/89e56e8fe7c08923a10c3a3dcdba90924a3eb1b8/pilot/test/integration/infra.go#L617-L626

Reply:
Definitely seems like something worth exploring ... care to file a bug and assign to me?